patches/minecraft/net/minecraft/server/level/ServerPlayer.java.patch

```diff
@@ -134,11 +134,12 @@
     }
  
     protected void m_142540_(MobEffectInstance p_143393_, @Nullable Entity p_143394_) {
-@@ -1181,6 +_,7 @@
+@@ -1181,6 +_,8 @@
     }
  
     public boolean m_143403_(GameType p_143404_) {
-+      if (!net.minecraftforge.common.ForgeHooks.onChangeGameMode(this, this.f_8941_.m_9290_(), p_143404_)) return false;
++      p_143404_ = net.minecraftforge.common.ForgeHooks.onChangeGameType(this, this.f_8941_.m_9290_(), p_143404_);
```

Comment:
This should be onChangeGameMode (the boolean), not onChangeGameType (the Game Type)

Reply:
But then you can't set the parameter?

Reply:
No it should not...? onChangeGameMode is the old, deprecated one.
A boolean is not enough as a return type, because the event can change the new gametype.

Reply:
Oh I see, I misunderstood the next line. Ignore my comment, then.